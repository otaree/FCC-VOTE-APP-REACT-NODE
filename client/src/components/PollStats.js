import React from 'react';

const PollStats = props => {
    return (
        <div>
            <ul>
                {
                    props.poll.options.map(option => <li key={option._id} >{option.value}:{option.votes}</li>)
                }
            </ul>
        </div>
    );
};

export default PollStats;