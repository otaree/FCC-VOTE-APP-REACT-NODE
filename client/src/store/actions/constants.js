// auth
export const AUTH_LOADING = "AUTH_LOADING";
export const AUTH_GET_UID = "AUTH_GET_UID";
export const AUTH_LOGIN = "AUTH_LOGIN";
export const AUTH_LOGOUT = "AUTH_LOGOUT";
export const AUTH_FAIL = "AUTH_FAIL";

// poll
export const POLL_LOADING = "POLL_LOADING";
export const POLL_SUCCESS = "POLL_SUCCESS";
export const POLL_FAIL = "POLL_FAIL";
export const POLL_SET_POLLS = "POLL_SET_POLLS";
export const POLL_SET = "POLL_SET";
export const POLL_UNSET = "POLL_UNSET";