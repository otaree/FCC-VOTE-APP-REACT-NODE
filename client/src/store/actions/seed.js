export const polls = [
    {
        _id: "1",
        question: "Yes or No?",
        createdAt: 1000,
        author: '123abc',
        voters: ["123abc"],
        options: [{
            votes: 1,
            value: 'Yes',
            _id: "101"
        }, {
            votes: 0,
            value: 'No',
            _id: "102"
        }]
    },
    {
        _id: "2",
        question: "2017 or 2018?",
        createdAt: 2000,
        author: '124abc',
        voters: ["124abc"],
        options: [{
            votes: 1,
            value: '2018',
            _id: "201"
        }, {
            votes: 0,
            value: '2017',
            _id: "202"
        }]
    }
];