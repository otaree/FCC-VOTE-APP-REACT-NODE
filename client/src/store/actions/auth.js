import axios from 'axios';
import uniqid from 'uniqid';

import * as constants from './constants';

export const loading = () => {
    return {
        type: constants.AUTH_LOADING,
        loading: true
    };
};

export const login = (user) => {
    return {
        ...user,
        loading: false,
        error: {
            isError: false,
            value: ''
        },
        type: constants.AUTH_LOGIN
    }
};

export const logout = () => {
    return {
        type: constants.AUTH_LOGOUT,
        loading: false,
        error: {
            isError: false,
            value: ''
        }
    };
};

export const fail = (value) => {
    return {
        type: constants.AUTH_FAIL,
        loading: false,
        error: {
            isError: true,
            value
        }
    };
};

export const initUser = (token) => {
    return async dispatch => {
        dispatch(loading());
        try {
            const response = await axios({ url: 'http://localhost:5000/user/token', method: 'get', headers: { 'x-auth': token }});
            dispatch(login(response.data));
        } catch (e) {
            dispatch(fail("Authentication fail"));
        }
    };
};

export const getUid = () => {
    let uid = localStorage.getItem("uid");
    if (!uid) {
        uid = uniqid();
        localStorage.setItem("uid", uid)
    }
    return {
        type: constants.AUTH_GET_UID,
        uid
    }
};