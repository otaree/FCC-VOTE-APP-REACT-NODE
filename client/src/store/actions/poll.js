import axios from 'axios';

import * as constants from './constants';

export const pollLoading = () => ({
    type: constants.POLL_LOADING,
    loading: true
});

export const pollSuccess = () => ({
    type: constants.POLL_SUCCESS,
    loading: false,
    error: {
        isError: false,
        value: ''
    }
});

export const pollFail = value => ({
    type: constants.POLL_FAIL,
    loading: false,
    error: {
        isError: true,
        value
    }
});

export const setPolls = polls => ({
    type: constants.POLL_SET_POLLS,
    loading: false,
    error: {
        isError: false,
        value: ''
    },
    polls
});

export const setPoll = poll => ({
    type: constants.POLL_SET,
    loading: false,
    error: {
        isError: false,
        value: ''
    },
    poll
});

export const fetchPolls = () => {
    return async dispatch => {
        dispatch(pollLoading());

        try {
            const response = await axios.get('http://localhost:5000/polls');
            dispatch(setPolls(response.data));
            return Promise.resolve("PASS");
        } catch (e) {
            dispatch(pollFail("Network Problem"));
            return Promise.reject("FAIL");
        }
    }
};

export const fetchPoll = (id) => {
    return async dispatch => {
        dispatch(pollLoading());

        try {
            const response = await axios.get(`http://localhost:5000/poll/${id}`);
            dispatch(setPoll(response.data));
            return Promise.resolve();
        } catch (e) {
            dispatch(pollFail("Network Problem"));
            return Promise.reject();
        }
    };
};

export const createPoll = (poll, token) => {
    return async dispatch => {
        dispatch(pollLoading());

        try {
            const response = await axios({ url: "http://localhost:5000/poll", method: "post", headers: { "x-auth": token }, data: poll });
            dispatch(pollSuccess());
            return response.data;
        } catch (e) {
            dispatch(pollFail("Network Problem"));
            return Promise.reject();
        }
    }
};

export const updatePoll = (poll, token, id) => {
    return async dispatch => {
        dispatch(pollLoading());

        try {
            const response = await axios({ url: `http://localhost:5000/poll/${id}`, method: "patch", headers: { "x-auth": token }, data: poll });
            dispatch(pollSuccess());
            return response.data;
        } catch (e) {
            dispatch(pollFail("Network Problem"));
            return Promise.reject();
        }
    }
};

export const deletePoll = (token, id) => {
    return async dispatch => {
        dispatch(pollLoading());

        try {
            const response = await axios({ url: `http://localhost:5000/poll/${id}`, method: "delete", headers: { "x-auth": token } });
            dispatch(pollSuccess());
            return response.data;
        } catch (e) {
            dispatch(pollFail("Network Problem"));
            return Promise.reject();
        }
    }
};

export const votePoll = (id, optionId, userId) => {
    return async dispatch => {
        dispatch(pollLoading());

        try {
            const response = await axios({ url: `http://localhost:5000/poll/${id}/vote`, method: "patch", data: { option: optionId, userId } });
            dispatch(pollSuccess());
            return response.data;
        } catch (e) {
            dispatch(pollFail("Network Problem"));
            return Promise.reject();
        }
    }
};