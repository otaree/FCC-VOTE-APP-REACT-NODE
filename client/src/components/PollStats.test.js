import React from 'react';
import { shallow } from 'enzyme';

import PollStats from './PollStats';
import { polls } from '../store/actions/seed';

describe("PollStats", () => {
    const props = {
        poll: polls[0]
    };

    const pollStats = shallow(<PollStats {...props} />);

    it('renders correctly', () => {
        expect(pollStats).toMatchSnapshot();
    });
});