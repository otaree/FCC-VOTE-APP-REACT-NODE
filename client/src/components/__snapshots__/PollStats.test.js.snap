// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollStats renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PollStats
    poll={
        Object {
            "_id": "1",
            "author": "123abc",
            "createdAt": 1000,
            "options": Array [
              Object {
                "_id": "101",
                "value": "Yes",
                "votes": 1,
              },
              Object {
                "_id": "102",
                "value": "No",
                "votes": 0,
              },
            ],
            "question": "Yes or No?",
            "voters": Array [
              "123abc",
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Yes or No?
</h2>,
        <p>
          by 
          123abc
</p>,
        <ul>
          <li>
                    Yes
                    :
                    1
          </li>
          <li>
                    No
                    :
                    0
          </li>
</ul>,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Yes or No?",
        },
        "ref": null,
        "rendered": "Yes or No?",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "by ",
            "123abc",
          ],
        },
        "ref": null,
        "rendered": Array [
          "by ",
          "123abc",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              Yes
              :
              1
</li>,
            <li>
              No
              :
              0
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "101",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Yes",
                ":",
                1,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Yes",
              ":",
              1,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "102",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "No",
                ":",
                0,
              ],
            },
            "ref": null,
            "rendered": Array [
              "No",
              ":",
              0,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Yes or No?
</h2>,
          <p>
            by 
            123abc
</p>,
          <ul>
            <li>
                        Yes
                        :
                        1
            </li>
            <li>
                        No
                        :
                        0
            </li>
</ul>,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Yes or No?",
          },
          "ref": null,
          "rendered": "Yes or No?",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "by ",
              "123abc",
            ],
          },
          "ref": null,
          "rendered": Array [
            "by ",
            "123abc",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                Yes
                :
                1
</li>,
              <li>
                No
                :
                0
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "101",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Yes",
                  ":",
                  1,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Yes",
                ":",
                1,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "102",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "No",
                  ":",
                  0,
                ],
              },
              "ref": null,
              "rendered": Array [
                "No",
                ":",
                0,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
